### PR TITLE
Add devenv configuration

### PR DESCRIPTION
**Pull request by straight-shoota**
Configuration for a https://devenv.sh environment with all necessary tools to build and test shards.
I'm using this locally and it makes sense to share it in the repo so others can make use of it as well.

### CODE REVIEW

**Comment by straight-shoota** [comment]
Potentially, this can be useful for setting up an environment in CI as well.
However our current setup works just fine across multiple platforms (including Windows which Nix does not support), so it's fine for the moment.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1 @@
+/nix/store/2bvs1hdl54w2x8wjbh2icvv6wqn69dzj-pre-commit-config.json
```
Comment (Sija) [inline .pre-commit-config.yaml:1]: Is this file needed?

Comment (straight-shoota) [reply .pre-commit-config.yaml:1]: Uh yeah, apparently not. Good catch!
Dunno if this has changed since I originally ran `devenv init` (that was a while ago). Nowadays it puts this file in `.gitignore`. So I'll just rebuild from scratch to get things setup to the latest configuration.

**Review by beta-ziliani — APPROVED** [review]
